transformers/events/pot_file/dsr/transformer.go

```diff
@@ -8,15 +8,23 @@ import (
 	"github.com/makerdao/vulcanizedb/pkg/datastore/postgres"
 )
 
+// Transformer implements the VDB event Transformer interface
 type Transformer struct{}
 
-func (Transformer) ToModels(_ string, logs []core.EventLog, _ *postgres.DB) ([]event.InsertionModel, error) {
+// ToModels transforms log data into general InsertionModels the Repository can persist
+func (Transformer) ToModels(_ string, logs []core.EventLog, db *postgres.DB) ([]event.InsertionModel, error) {
 	var results []event.InsertionModel
 	for _, log := range logs {
 		verifyErr := shared.VerifyLog(log.Log, shared.FourTopicsRequired, shared.LogDataNotRequired)
 		if verifyErr != nil {
 			return nil, verifyErr
 		}
+		msgSender := shared.GetChecksumAddressString(log.Log.Topics[1].Hex())
+		msgSenderID, msgSenderErr := shared.GetOrCreateAddress(msgSender, db)
+		if msgSenderErr != nil {
+			return nil, msgSenderErr
```

Comment:
not sure if the value add is worth revising, but I think in a lot of places we use `shared.ErrCouldNotCreateFK` to wrap errors like this